data/src/main/java/io/gnosis/data/models/Transaction.kt

```diff
@@ -1,11 +1,157 @@
 package io.gnosis.data.models
 
-sealed class Transaction(open val nonce: String) {
-
-    data class ChangeMastercopy(override val nonce: String) : Transaction(nonce)
-    data class ChangeMastercopyQueued(override val nonce: String) : Transaction(nonce)
-    data class SettingsChange(override val nonce: String) : Transaction(nonce)
-    data class SettingsChangeQueued(override val nonce: String) : Transaction(nonce)
-    data class Transfer(override val nonce: String) : Transaction(nonce)
-    data class TransferQueued(override val nonce: String) : Transaction(nonce)
+import io.gnosis.data.backend.dto.ServiceTokenInfo
+import pm.gnosis.model.Solidity
+import java.math.BigInteger
+
+sealed class Transaction {
+    data class Custom(
+        val nonce: BigInteger?,
+        val address: Solidity.Address,
+        val dataSize: Long,
+        val date: String?,
+        val value: BigInteger
+    ) : Transaction()
+
+    data class SettingsChange(
+        val dataDecoded: DataDecodedDto,
+        val date: String?,
+        val nonce: BigInteger
+    ) : Transaction()
+
+    data class Transfer(
+        val recipient: Solidity.Address,
+        val sender: Solidity.Address,
+        val value: BigInteger,
+        val date: String?,
+        val tokenInfo: ServiceTokenInfo
+    ) : Transaction()
+}
+
+interface TransactionDto {
+    val txType: TransactionType?
+    val to: Solidity.Address
+    val data: String?
+}
+
+data class MultisigTransactionDto(
```

Comment:
So all Dtos go to the dto package. And where do the enums go?

Reply:
depends whether those are dto or model enums. If they are used in dto -> then dto. Otherwise models. We should try to keep our dto's & models separate

Reply:
Done in a9accbb006d6fba824b8d4ebea9fd6272fb899ea